tokio/src/runtime/coop.rs

```diff
@@ -240,6 +243,42 @@ cfg_coop! {
             self.0.is_none()
         }
     }
+
+    pin_project! {
+        /// A future type that calls `poll_proceed` before polling the inner future to check if the
+        /// inner future has exceeded its budget. If the inner future resolves, this will
+        /// automatically call `RestoreOnPending::made_progress` before resolving this future with
+        /// the result of the inner one. If polling the inner future is pending, polling this future
+        /// type will also return a `Poll::Pending`.
```

Comment:
This isn't publicly visible, so it's not a big deal here, but generally the convention is that documentation should start with one *short* line. If more explanation is needed, it can go in a separate paragraph. This is because when structs are shown in documentation, the first line is used as a summary in some cases, and the summary should be short.